include/Piece.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include "macros.h"
+
+class Piece {
+
+public:
+	Piece(bool isWhite);
+	virtual ~Piece() = default;

```

Comment:
No need for this line, compiler will usually add the default d'tor for you if you don't mention it